File: frontend/src/types/medicalRecord.ts
/**
 * Medical Record TypeScript types
 */

export interface MedicalRecord {
  id: string;
  pet_id: string;
  version: number;
  is_current: boolean;
  record_type: string;
  title: string;
  description?: string;
  medical_data?: Record<string, any>;
  visit_date: string; // ISO date string
  veterinarian_name?: string;
  clinic_name?: string;
  diagnosis?: string;
  treatment?: string;
  medications?: string;
  follow_up_required: boolean;
  follow_up_date?: string; // ISO date string
  weight?: number;
  temperature?: number;
  cost?: number;
  created_by_user_id: string;
  created_at: string;
  updated_at: string;
  is_follow_up_due: boolean;
  days_since_visit: number;
}

export interface MedicalRecordWithRelations extends MedicalRecord {
  pet: {
    id: string;
    name: string;
    species: string;
    breed?: string;
  };
  created_by: {
    id: string;
    full_name?: string;
    email?: string;
  };
}

export interface MedicalRecordFormData {
  record_type: string;
  title: string;
  description?: string;
  medical_data?: Record<string, any>;
  visit_date: string;
  veterinarian_name?: string;
  clinic_name?: string;
  diagnosis?: string;
  treatment?: string;
  medications?: string;
  follow_up_required: boolean;
  follow_up_date?: string;
  weight?: number;
  temperature?: number;
  cost?: number;
}

export interface MedicalRecordCreateRequest extends MedicalRecordFormData {
  pet_id: string;
}

export interface MedicalRecordUpdateRequest extends Partial<MedicalRecordFormData> {}

export interface MedicalRecordListResponse {
  records: MedicalRecord[];
  total: number;
  current_records_count: number;
  historical_records_count: number;
}

export interface MedicalRecordTimelineResponse {
  pet_id: string;
  pet_name: string;
  records_by_date: MedicalRecord[];
  follow_up_due: MedicalRecord[];
  recent_weight?: number;
  weight_trend?: string;
}

export const MEDICAL_RECORD_TYPES = [
  'checkup',
  'vaccination',
  'surgery',
  'emergency',
  'dental',
  'grooming',
  'diagnostic',
  'treatment',
  'follow_up',
  'other'
] as const;

export type MedicalRecordType = typeof MEDICAL_RECORD_TYPES[number];

export const MEDICAL_RECORD_TYPE_LABELS: Record<MedicalRecordType, string> = {
  checkup: 'Regular Checkup',
  vaccination: 'Vaccination',
  surgery: 'Surgery',
  emergency: 'Emergency Visit',
  dental: 'Dental Care',
  grooming: 'Grooming',
  diagnostic: 'Diagnostic Test',
  treatment: 'Treatment',
  follow_up: 'Follow-up Visit',
  other: 'Other'
};

export const MEDICAL_RECORD_TYPE_COLORS: Record<MedicalRecordType, string> = {
  checkup: 'bg-blue-100 text-blue-800',
  vaccination: 'bg-green-100 text-green-800',
  surgery: 'bg-red-100 text-red-800',
  emergency: 'bg-red-100 text-red-800',
  dental: 'bg-yellow-100 text-yellow-800',
  grooming: 'bg-purple-100 text-purple-800',
  diagnostic: 'bg-indigo-100 text-indigo-800',
  treatment: 'bg-orange-100 text-orange-800',
  follow_up: 'bg-teal-100 text-teal-800',
  other: 'bg-gray-100 text-gray-800'
};

export const MEDICAL_DATA_TEMPLATES: Record<string, Record<string, string>> = {
  vaccination: {
    vaccine_name: '',
    vaccine_type: '',
    batch_number: '',
    expiration_date: '',
    next_due_date: '',
    adverse_reactions: ''
  },
  surgery: {
    procedure_name: '',
    anesthesia_type: '',
    complications: '',
    recovery_notes: '',
    suture_removal_date: '',
    activity_restrictions: ''
  },
  checkup: {
    heart_rate: '',
    respiratory_rate: '',
    blood_pressure: '',
    body_condition_score: '',
    dental_condition: '',
    skin_condition: '',
    eye_condition: '',
    ear_condition: ''
  },
  diagnostic: {
    test_type: '',
    test_results: '',
    reference_ranges: '',
    abnormal_findings: '',
    recommendations: ''
  }
};

File: iphone/HelpPetAI/HelpPetAI/Models/MedicalRecord.swift
//
//  MedicalRecord.swift
//  HelpPetAI
//
//  Created by Justin Zollars on 9/1/25.
//

// MARK: - 4. Models/MedicalRecord.swift
import Foundation

struct MedicalRecord: Codable, Identifiable {
    let id: String
    let petId: String
    let recordType: String
    let title: String
    let description: String?
    let medicalData: MedicalData?
    let visitDate: Date
    let veterinarianName: String?
    let clinicName: String?
    let diagnosis: String?
    let treatment: String?
    let medications: String?
    let followUpRequired: Bool
    let followUpDate: Date?
    let weight: Double?
    let temperature: Double?
    let cost: Double?
    let version: Int
    let isCurrent: Bool
    let createdByUserId: String
    let createdAt: Date
    let updatedAt: Date
    let isFollowUpDue: Bool
    let daysSinceVisit: Int
    
    enum CodingKeys: String, CodingKey {
        case id, title, description, diagnosis, treatment, medications, weight, temperature, cost, version
        case petId = "pet_id"
        case recordType = "record_type"
        case medicalData = "medical_data"
        case visitDate = "visit_date"
        case veterinarianName = "veterinarian_name"
        case clinicName = "clinic_name"
        case followUpRequired = "follow_up_required"
        case followUpDate = "follow_up_date"
        case isCurrent = "is_current"
        case createdByUserId = "created_by_user_id"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case isFollowUpDue = "is_follow_up_due"
        case daysSinceVisit = "days_since_visit"
    }
}

struct MedicalData: Codable {
    let heartRate: String?
    let respiratoryRate: String?
    let bloodPressure: String?
    let bodyConditionScore: String?
    let dentalCondition: String?
    let skinCondition: String?
    let eyeCondition: String?
    let earCondition: String?
    
    enum CodingKeys: String, CodingKey {
        case heartRate = "heart_rate"
        case respiratoryRate = "respiratory_rate"
        case bloodPressure = "blood_pressure"
        case bodyConditionScore = "body_condition_score"
        case dentalCondition = "dental_condition"
        case skinCondition = "skin_condition"
        case eyeCondition = "eye_condition"
        case earCondition = "ear_condition"
    }
}

struct MedicalRecordsResponse: Codable {
    let records: [MedicalRecord]
    let total: Int
    let currentRecordsCount: Int
    let historicalRecordsCount: Int
    
    enum CodingKeys: String, CodingKey {
        case records, total
        case currentRecordsCount = "current_records_count"
        case historicalRecordsCount = "historical_records_count"
    }
}

struct CreateMedicalRecordRequest: Codable {
    let recordType: String
    let title: String
    let description: String?
    let medicalData: MedicalData?
    let visitDate: Date
    let veterinarianName: String?
    let clinicName: String?
    let diagnosis: String?
    let treatment: String?
    let medications: String?
    let followUpRequired: Bool
    let followUpDate: Date?
    let weight: Double?
    let temperature: Double?
    let cost: Double?
    let petId: String
    
    enum CodingKeys: String, CodingKey {
        case title, description, diagnosis, treatment, medications, weight, temperature, cost
        case recordType = "record_type"
        case medicalData = "medical_data"
        case visitDate = "visit_date"
        case veterinarianName = "veterinarian_name"
        case clinicName = "clinic_name"
        case followUpRequired = "follow_up_required"
        case followUpDate = "follow_up_date"
        case petId = "pet_id"
    }
}

struct UpdateMedicalRecordRequest: Codable {
    let recordType: String
    let title: String
    let description: String
    let medicalData: MedicalData
    let visitDate: Date
    let veterinarianName: String
    let clinicName: String
    let diagnosis: String
    let treatment: String
    let medications: String
    let followUpRequired: Bool
    let followUpDate: Date?
    let weight: Double?
    let temperature: Double?
    let cost: Double?
    
    enum CodingKeys: String, CodingKey {
        case title, description, diagnosis, treatment, medications, weight, temperature, cost
        case recordType = "record_type"
        case medicalData = "medical_data"
        case visitDate = "visit_date"
        case veterinarianName = "veterinarian_name"
        case clinicName = "clinic_name"
        case followUpRequired = "follow_up_required"
        case followUpDate = "follow_up_date"
    }
}
